components/server/src/ome/security/CompositeACLVoter.java

```diff
@@ -100,6 +101,21 @@ public void throwUpdateViolation(IObject object) throws SecurityViolation {
         return choose().restrictions(object);
     }
 
+    @Override
+    public void setPermittedClasses(Map<Integer, Set<Class<? extends IObject>>> objectClassesPermitted) {
+        basic.setPermittedClasses(objectClassesPermitted);
```

Comment:
We don't have a particular session context at the time this is called when the graph request factory sets up so `choose()` has no basis to act, and we know it's only the `BasicACLVoter` that we care about here. (Indeed, once we get rid of shares, a bunch of this can be much simplified!)

In general forcing light admin checks into the prior security system has been rather unfortunate in making dependencies messier. When we have the basic functionality working it would be good to look again at the bean architecture here to see where I could clean things up.